backend/routes.py

```
@@ -35,6 +37,96 @@ def register_event_handlers():
+            if not repo_url.startswith(("http://", "https://")):
+                repo_url = "https://" + repo_url
+            
+            # Extract path from URL
+            try:
+                # Parse URL to get path component
+                if "github.com/" in repo_url:
+                    path = repo_url.split("github.com/", 1)[1]
+                else:
+                    raise HTTPException(status_code=400, detail="Invalid GitHub repository URL")
+            except (IndexError, ValueError):
+                raise HTTPException(status_code=400, detail="Invalid GitHub repository URL")
```

Ruff (0.14.11)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)